Recruit-Training/Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,15 +58,30 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+
     QStringList sections = line.split(BATDATA_DELIMITER);
 
+    if(sections.length() != COLUMNS){
+        return false;
+    }
+
     QString timeString = sections.at(0);
-    batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
 
-    batteryData.current = sections.at(2).toDouble();
 
-    return true;
+    QTime time1 = QTime::fromString(timeString, STRING_TIME_FORMAT);
+    batteryData.time = time1;
```

Comment:
You're not doing anything with time1, so it'll be cleaner to just do `batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);`